[CI] ModelPlotsIT.testPartitionFieldWithTerms

Log:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-unix-compatibility/os=opensuse/1066/console

```
  2> REPRODUCE WITH: ./gradlew :x-pack:qa:ml-native-tests:integTestRunner -Dtests.seed=7B45534D9707C727 -Dtests.class=org.elasticsearch.xpack.ml.integration.ModelPlotsIT -Dtests.method="testPartitionFieldWithTerms" -Dtests.security.manager=true -Dtests.locale=en-PH -Dtests.timezone=Antarctica/Palmer
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <23>
   >      but: was <24>
   >   at __randomizedtesting.SeedInfo.seed([7B45534D9707C727:3650B455272ED35]:0)
   >   at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >   at org.elasticsearch.xpack.ml.integration.ModelPlotsIT.testPartitionFieldWithTerms(ModelPlotsIT.java:104)
   >   at java.lang.Thread.run(Thread.java:748)
```

@elastic/ml-core could you please have a look ?

--- a/x-pack/qa/ml-native-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ModelPlotsIT.java
+++ b/x-pack/qa/ml-native-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ModelPlotsIT.java
@@ -30,7 +30,9 @@ import java.util.stream.Collectors;
 
 import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
 
 public class ModelPlotsIT extends MlNativeAutodetectIntegTestCase {
 
@@ -83,7 +85,11 @@ public class ModelPlotsIT extends MlNativeAutodetectIntegTestCase {
         startDatafeed(datafeedId, 0, System.currentTimeMillis());
         waitUntilJobIsClosed(job.getId());
 
-        assertThat(getBuckets(job.getId()).size(), equalTo(23));
+        // As the initial time is random, there's a chance the first record is
+        // aligned on a bucket start. Thus we check the buckets are in [23, 24]
+        assertThat(getBuckets(job.getId()).size(), greaterThanOrEqualTo(23));
+        assertThat(getBuckets(job.getId()).size(), lessThanOrEqualTo(24));
+
         Set<String> modelPlotTerms = modelPlotTerms(job.getId(), "partition_field_value");
         assertThat(modelPlotTerms, containsInAnyOrder("user_1", "user_2", "user_3"));
     }
@@ -101,7 +107,11 @@ public class ModelPlotsIT extends MlNativeAutodetectIntegTestCase {
         startDatafeed(datafeedId, 0, System.currentTimeMillis());
         waitUntilJobIsClosed(job.getId());
 
-        assertThat(getBuckets(job.getId()).size(), equalTo(23));
+        // As the initial time is random, there's a chance the first record is
+        // aligned on a bucket start. Thus we check the buckets are in [23, 24]
+        assertThat(getBuckets(job.getId()).size(), greaterThanOrEqualTo(23));
+        assertThat(getBuckets(job.getId()).size(), lessThanOrEqualTo(24));
+
         Set<String> modelPlotTerms = modelPlotTerms(job.getId(), "partition_field_value");
         assertThat(modelPlotTerms, containsInAnyOrder("user_2", "user_3"));
     }

